src/agent/communicator/src/communicator.cpp

```diff
@@ -4,26 +4,32 @@
 #include <boost/asio/ssl/stream.hpp>
 #include <boost/beast/version.hpp>
 #include <iostream>
-#include <nlohmann/json.hpp>
 #include <jwt-cpp/jwt.h>
+#include <nlohmann/json.hpp>
 
 using tcp = boost::asio::ip::tcp;
 using json = nlohmann::json;
 
 namespace communicator
 {
 
-    Communicator::Communicator()
-    : m_managerIp("localhost")
-    , m_port("8080")
+    Communicator::Communicator(const std::shared_ptr<iConfigurationParser>& configParser)
+        : m_managerIp("localhost")
+        , m_port("8080")
     {
-
+        if (configParser != nullptr)
+        {
+            m_managerIp = configParser->GetConfigString("agent", "manager_ip");
```

Comment:
We are setting these values twice, here and in line 17-18

Reply:
Pointer may be null, in fact that is the case for test builds, so in lines 17-18 we initialize these variables with default values.

Reply:
Removed.